src/column.rs

```diff
@@ -44,8 +44,8 @@ impl<'a, T: Component> Column<'a, T> {
             .entities
             .get(entity.id as usize)
             .map_or(true, |meta| meta.generation == entity.generation)
-            .then(|| self.entities[entity.id as usize])
-            .ok_or(NoSuchEntity)?;
+            .then(|| self.entities.get(entity.id as usize).ok_or(NoSuchEntity))
+            .ok_or(NoSuchEntity)??;
```

Comment:
I am really bad at functional style 😭

Reply:
Sorry, I skimmed this a little too quickly. The real (mostly cosmetic) issue here is that we're indexing into `entities` twice. A tidier approach would be:
```rust
let meta = self
    .entities
    .get(entity.id as usize)
    .filter(|meta| meta.generation == entity.generation)
    .ok_or(NoSuchEntity)?;
```